client/src/main/java/com/spectralogic/tpfr/client/model/ReWrapParams.java

```diff
@@ -0,0 +1,34 @@
+package com.spectralogic.tpfr.client.model;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+public class ReWrapParams {
+    private final Map<String, String> params;
+
+    /**
+     *
+     * @param filePath The full path to the media file whose partial offsets are being requested.
+     * @param tcin Timecode of the first frame requested
+     * @param tcout Timecode of the last frame requested
+     * @param fileFrameRate Frame rate, as returned in the file status report
+     * @param partFile Full UNC path to partial restored file fragment
+     * @param outFileName output file name for partial media file (care should be taken that this does not clash with other part restores, e.g. from other sections of the same source file). This should not have an extension, as this will added automatically.
+     */
+    public ReWrapParams(final String filePath, final TimeCode tcin, final TimeCode tcout, final String fileFrameRate,
+                        final String partFile, final String outFileName) {
+        params = new ImmutableMap.Builder<String, String>()
```

Comment:
You can use ImmutableMap.Of(...) up to 5 (k, v) pairs, here i have 6 so i have to use the builder